projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts

```diff
@@ -207,4 +245,48 @@ export class IgxGridGroupByRowComponent {
         const column = this.groupRow.column;
         return (column && column.dataType) || DataType.String;
     }
+
+    /**
+     * @hidden @internal
+     */
+    public get areAllRowsInTheGroupSelected(): boolean {
+        return this.groupRow.records.every(x => this.gridSelection.isRowSelected(x));
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get selectedRowsInTheGroup(): any[] {
+        return this.groupRow.records.filter(rowID => this.gridSelection.filteredSelectedRowIds.indexOf(rowID) > -1);
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get groupByRowCheckboxCheckedState(): boolean {
+        return this.selectedRowsInTheGroup.length === this.groupRow.records.length ? true : false;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get groupByRowCheckboxIndeterminateState(): boolean {
+        if (this.selectedRowsInTheGroup.length > 0) {
+            return this.selectedRowsInTheGroup.length !== this.groupRow.records.length ? true : false;
+        }
+        return false;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get groupByRowSelectorBaseAriaLabel() {
+        let ariaLabel: string = this.areAllRowsInTheGroupSelected ? 'Deselect all rows in the group with field name ' : 'Select all rows in the group with field name ';
```

Comment:
@zdrawku , @radomirchev - do we localize aria more elaborate labels as part of the ARIA efforts?

Reply:
@damyanpetev thinks we should localize such labels . @randriova should be able to help with the proper way to combine them .

Reply:
It's perhaps not as widespread as it should be right now, but never too late to start localizing those. If it's text that's meant to be visible or used by assistive tech, chances are it should come from the resource strings.